rotkehlchen/db/history_events.py

```diff
@@ -397,39 +397,65 @@ def get_history_events(
         TODO: To not query all columns with all joins for all cases, we perhaps can
         peek on the entry type of the filter and adjust the SELECT fields accordingly?
         """
+        bindings, type_idx, order_clause, final_rearrangement = [], 0, '', ''
         base_prefix = 'SELECT '
         base_suffix = f'{HISTORY_BASE_ENTRY_FIELDS}, {EVM_EVENT_FIELDS}, {ETH_STAKING_EVENT_FIELDS} {ALL_EVENTS_DATA_JOIN}'  # noqa: E501
-        bindings = []
-        type_idx = 1 if group_by_event_ids is True else 0
-        special_free_query = False
-        free_query_group_by = maybe_filter_ignore_asset(filter_query)
+        group_by_event_id_query = (
+            f'{base_prefix} COUNT(*), * FROM (SELECT {base_suffix} WHERE event_identifier '
+            'IN (SELECT DISTINCT event_identifier FROM history_events {inner_filter} {limit})) {outer_filter}'  # noqa: E501
+        )
+        ignore_asset_filter = maybe_filter_ignore_asset(filter_query)
+        if filter_query.order_by is not None and (has_premium or group_by_event_ids):
+            order_clause = f' {filter_query.order_by.prepare()}'
 
-        if has_premium is True:
-            if group_by_event_ids is True:
-                base_prefix += 'COUNT(*), '
-            base_query = f'{base_prefix} {base_suffix}'
-        else:
-            if group_by_event_ids is True:
-                special_free_query = True  # a bit ugly conditions to keep limit at groups for free users  # noqa: E501
-                free_query_group_by += 'GROUP BY event_identifier'
-                base_query = (
-                    f'{base_prefix} * FROM (SELECT COUNT(*), {base_suffix} '
-                    f'{free_query_group_by} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) '
+        if has_premium:
+            if group_by_event_ids:
+                filters, filters_bindings = filter_query.prepare(
+                    with_order=False,
+                    with_pagination=False,
+                    special_free_query=True,
```

Comment:
In the premium I believe we don´t use it since there are no limits. For free users we wanted to prevent them from getting no events due to spam

Reply:
yeah that was the reason we had this flag, but there is another reason to seperate ignored asset filter from other filters.

When excluding the ignored assets and enabling grouping, if we apply the ignored assets exclusion filter in the outer query along with other filters then the group having one event with ignored asset is also returned because other events of that group have normal assets.
So, now I'm eliminating them in the sub query for both cases, so that in the outer query none of the events are present which are part of a group in which at least one event have ignored event.

This is how the query is forming with current logic:
```sql
SELECT COUNT(*), * FROM (
    SELECT entry_type, history_events.identifier AS history_events_identifier, event_identifier, sequence_index, timestamp, location, location_label, asset, amount, usd_value, notes, type, subtype , tx_hash, counterparty, product, address, extra_data, validator_index, is_exit_or_blocknumber FROM history_events
    LEFT JOIN evm_events_info ON history_events.identifier=evm_events_info.identifier
    LEFT JOIN eth_staking_events_info ON history_events.identifier=eth_staking_events_info.identifier
    WHERE event_identifier NOT IN (
        SELECT event_identifier FROM history_events
        WHERE (asset IN (SELECT value FROM multisettings WHERE name="ignored_asset"))
    )
)
GROUP BY event_identifier
ORDER BY timestamp ASC,sequence_index ASC
```

This is formed when doing it along with the filters:
```sql
SELECT COUNT(*), * FROM (
    SELECT entry_type, history_events.identifier AS history_events_identifier, event_identifier, sequence_index, timestamp, location, location_label, asset, amount, usd_value, notes, type, subtype , tx_hash, counterparty, product, address, extra_data, validator_index, is_exit_or_blocknumber FROM history_events
    LEFT JOIN evm_events_info ON history_events.identifier=evm_events_info.identifier
    LEFT JOIN eth_staking_events_info ON history_events.identifier=eth_staking_events_info.identifier
)
WHERE (asset IS NULL OR asset NOT IN (SELECT value FROM multisettings WHERE name="ignored_asset"))
GROUP BY event_identifier
ORDER BY timestamp ASC,sequence_index ASC
```